content/en/docs/overview/pointers.md

```diff
@@ -3,12 +3,83 @@ title: "Pointers"
 weight: 230
 ---
 
-## Goals
+## General
 
-* TODO
+In addition to the basic data types Go also have pointers.
+A pointer contains a memory address of an actual value or `nil` if they do not point to anything.
+The zero value of a pointer is `nil`.
 
+Pointer types are indicated with a star.
+For example:
 
-## Pointers
+* The type `*int` is a pointer to an `int`
+* The type `*bool` is a pointer to a `bool`
+
+Concerning pointers you mainly have to remember two operations. How to create a pointer to a value and how go obtain a value from a pointer:
+
+| Operation | Example | Code | Description |
+| - | - | - | - |
+| **Value To Pointer** | `int` to `*int` | `myPointer = &myInt` | With the `&` operator you can create a pointer to a value |
+| **Value From Pointer** | `*int` to `int` | `myInt = *myPointer` | With the `*` operator we can obtain the actual value from a pointer (**dereference**)
+
+
+```golang
+package main
+
+import "fmt"
+
+func main() {
+	var (
+		// myPointer initially is nil (zero value)
+		myPointer *int
+	)
+
+	// myInt is of type int
+	myInt := 42
+
+	// Value To Pointer
+	// myPointer becomes pointer to myInt
+	myPointer = &myInt
+
+	// Value From Pointer
+	// Change myInt through myPointer. We can not assign an int to a *int hence we
+	// have to dereference myPointer with * to assign a new number
+	*myPointer = 12
+
+	fmt.Println(myInt)
+}
+<!--output-->
+12
+```
+
+If you dereference a pointer you should always be sure that it is not `nil`. Otherwise your program crashes with an error `invalid memory address or nil pointer dereference` (same as NullPointerException in Java).
+
+
+## Pointers As Function Arguments
+
+With pointers we can pass values to a function by reference.
```

Comment:
In Go function arguments are normally passed as values. This means they are copied in memory and a new variable is created. This can be a performance problem for large variables (e.g. structs). To overcome this we can use pointers. With pointers only the memory address to the variable value is passed and the variable itself is not copied. This is called passing values by reference. 

**Note:** One exception to the rule are slices. slices are always passed by reference. This has to do with the implementation of slices themselves, but it is not necessary to know the exact details.

When values are passed by reference, we can change the values within functions. Variables don't have to be returned.

Reply:
Some standard library functions expect pointers, but usually we do not have to use them. The most common case of pointers is with struct types, which we will learn about in the next lab.